holmes.go

```diff
@@ -169,7 +169,10 @@ func (h *Holmes) goroutineCheckAndDump(gNum int) {
 	if !h.opts.GrOpts.Enable {
 		return
 	}
-
+	if h.opts.GrOpts.GoroutineNumberMax != 0 && gNum >= h.opts.GrOpts.GoroutineNumberMax {
```

Comment:
better sample the max number of Goroutine instead of skipping it simply.

But seems pprof does not support sample Goroutine after I have checked its implementation.
Let's add a TODO comment here.